### PR TITLE
[docs] Update custom hooks for better understanding

### PR DESCRIPTION
**Pull request by LuckyPigeon**
Add some annotations and update `focusAtom` & `selectAtom` utility function with `useCallBack.

### CODE REVIEW

File: docs/advanced-recipes/custom-useatom-hooks.mdx [hunk 1]
```diff
@@ -14,12 +14,16 @@ This page shows the ways of creating different utility functions. Utility functi
 import { useAtom } from 'jotai'
 import { selectAtom } from 'jotai/utils'
 
+/* if an atom is created here, please use `useMemo(() => atom(initValue), [initValue])` instead. */
 export function useSelectAtom(anAtom, keyFn) {
-    return useAtom(selectAtom(anAtom, keyFn));
+    return useAtom(selectAtom(
+        anAtom, 
+        useCallBack(keyFn, [])
```
Comment (dai-shi) [inline docs/advanced-recipes/custom-useatom-hooks.mdx:21]: This is not a good example. Please make it realistic, something like:
```js
useCallback((state) => state[prop], [prop])
```

File: docs/advanced-recipes/custom-useatom-hooks.mdx [hunk 2]
```diff
@@ -51,9 +55,13 @@ export function useSplitAtom(anAtom) {
 import { useAtom } from 'jotai'
 import { focusAtom } from 'jotai/optics'
 
-export function useFocusAtom(anAtom, keyFn) {
-    return useAtom(focusAtom(anAtom, keyFn))
+/* if an atom is created here, please use `useMemo(() => atom(initValue), [initValue])` instead. */
+export function useFocusAtom(anAtom, (optic) => optic.prop('key')) {
```
Comment (dai-shi) [inline docs/advanced-recipes/custom-useatom-hooks.mdx:59]: leftover?
```suggestion
export function useFocusAtom(anAtom) {

```

File: docs/advanced-recipes/custom-useatom-hooks.mdx [hunk 3]
```diff
@@ -51,9 +55,13 @@ export function useSplitAtom(anAtom) {
 import { useAtom } from 'jotai'
 import { focusAtom } from 'jotai/optics'
 
-export function useFocusAtom(anAtom, keyFn) {
-    return useAtom(focusAtom(anAtom, keyFn))
+/* if an atom is created here, please use `useMemo(() => atom(initValue), [initValue])` instead. */
+export function useFocusAtom(anAtom, (optic) => optic.prop('key')) {
+    return useAtom(focusAtom(
+        anAtom,
+        useCallBack((optic) => optic.prop('key'), [])
```
Comment (dai-shi) [inline docs/advanced-recipes/custom-useatom-hooks.mdx:62]: This looks good.

**Review by dai-shi — CHANGES_REQUESTED** [review]
thanks for working on this. please see comments.

File: docs/advanced-recipes/custom-useatom-hooks.mdx (hunk 1 above)
Comment (LuckyPigeon) [reply docs/advanced-recipes/custom-useatom-hooks.mdx:21]: How about `prop` with single quote
`useCallback((state) => state['prop'], [])`

File: docs/advanced-recipes/custom-useatom-hooks.mdx [hunk 4]
```diff
@@ -14,12 +14,16 @@ This page shows the ways of creating different utility functions. Utility functi
 import { useAtom } from 'jotai'
 import { selectAtom } from 'jotai/utils'
 
-export function useSelectAtom(anAtom, keyFn) {
-    return useAtom(selectAtom(anAtom, keyFn));
+/* if an atom is created here, please use `useMemo(() => atom(initValue), [initValue])` instead. */
```
Comment (dai-shi) [inline docs/advanced-recipes/custom-useatom-hooks.mdx:17]: Hmm, on second thought, changing `useSelectAtom` is not the intention of the original recipe.

So, how about keeping the original function, and adding a usage code snippet?
Something like this:
```js
export function useSelectAtom(anAtom, keyFn) {
    return useAtom(selectAtom(anAtom, keyFn));
}

// how to use it
useSelectAtom(
  useMemo(() => atom(initValue), [initValue]),
  useCallBack((state) => state.prop, [])
)
```

Same for `useFocusAtom`.

Comment (LuckyPigeon) [reply docs/advanced-recipes/custom-useatom-hooks.mdx:17]: This one is better

File: docs/advanced-recipes/custom-useatom-hooks.mdx [hunk 5]
```diff
@@ -14,12 +14,19 @@ This page shows the ways of creating different utility functions. Utility functi
 import { useAtom } from 'jotai'
```
Comment (dai-shi) [inline docs/advanced-recipes/custom-useatom-hooks.mdx:14]:
```suggestion
import { useAtomValue } from 'jotai'

```

File: docs/advanced-recipes/custom-useatom-hooks.mdx [hunk 6]
```diff
@@ -51,9 +58,17 @@ export function useSplitAtom(anAtom) {
 import { useAtom } from 'jotai'
 import { focusAtom } from 'jotai/optics'
 
-export function useFocusAtom(anAtom, keyFn) {
-    return useAtom(focusAtom(anAtom, keyFn))
+/* if an atom is created here, please use `useMemo(() => atom(initValue), [initValue])` instead. */
+export function useSelectAtom(anAtom, keyFn) {
```
Comment (dai-shi) [inline docs/advanced-recipes/custom-useatom-hooks.mdx:63]:
```suggestion
export function useFocusAtom(anAtom, keyFn) {

```

File: docs/advanced-recipes/custom-useatom-hooks.mdx [hunk 7]
```diff
@@ -11,15 +11,23 @@ This page shows the ways of creating different utility functions. Utility functi
...
 ```ts
-import { useAtom } from 'jotai'
+import { useAtomValue } from 'jotai'
+
 import { selectAtom } from 'jotai/utils'
 
+/* if an atom is created here, please use `useMemo(() => atom(initValue), [initValue])` instead. */
 export function useSelectAtom(anAtom, keyFn) {
-    return useAtom(selectAtom(anAtom, keyFn));
+    return useAtomValue(selectAtom(anAtom, keyFn));
```
Comment (dai-shi) [inline docs/advanced-recipes/custom-useatom-hooks.mdx:20]:
```suggestion
    return useAtomValue(selectAtom(anAtom, keyFn))

```

File: docs/advanced-recipes/custom-useatom-hooks.mdx [hunk 8]
```diff
@@ -51,9 +59,18 @@ export function useSplitAtom(anAtom) {
 import { useAtom } from 'jotai'
 import { focusAtom } from 'jotai/optics'
 
+/* if an atom is created here, please use `useMemo(() => atom(initValue), [initValue])` instead. */
 export function useFocusAtom(anAtom, keyFn) {
-    return useAtom(focusAtom(anAtom, keyFn))
+
+    return useAtom(focusAtom(anAtom, keyFn));
```
Comment (dai-shi) [inline docs/advanced-recipes/custom-useatom-hooks.mdx:65]:
```suggestion
    return useAtom(focusAtom(anAtom, keyFn))
```

File: docs/advanced-recipes/custom-useatom-hooks.mdx [hunk 9]
```diff
@@ -11,15 +11,23 @@ This page shows the ways of creating different utility functions. Utility functi
 ### useSelectAtom
 
 ```ts
-import { useAtom } from 'jotai'
+import { useAtomValue } from 'jotai'
+
```
Comment (dai-shi) [inline docs/advanced-recipes/custom-useatom-hooks.mdx:15]:
```suggestion
import { useAtomValue } from 'jotai'
```

Comment (dai-shi) [inline docs/advanced-recipes/custom-useatom-hooks.mdx:15]:
```suggestion
```

File: docs/advanced-recipes/custom-useatom-hooks.mdx [hunk 10]
```diff
@@ -51,9 +58,17 @@ export function useSplitAtom(anAtom) {
 import { useAtom } from 'jotai'
 import { focusAtom } from 'jotai/optics'
 
+/* if an atom is created here, please use `useMemo(() => atom(initValue), [initValue])` instead. */
 export function useFocusAtom(anAtom, keyFn) {
     return useAtom(focusAtom(anAtom, keyFn))
 }
+
+// how to use it
+
```
Comment (dai-shi) [inline docs/advanced-recipes/custom-useatom-hooks.mdx:67]:
```suggestion
```

**Review by dai-shi — APPROVED** [review]
Merging this. Thanks for your contribution!

**Pull request merged**